broadcast/internal/arc/arc_submit_tx.go

```diff
@@ -139,21 +143,39 @@ func submitBatchTransactions(ctx context.Context, arc *ArcClient, txs []*broadca
 	return model, nil
 }
 
-func createSubmitTxBody(tx *broadcast.Transaction) ([]byte, error) {
-	body := &SubmitTxRequest{tx.RawTx}
-	data, err := json.Marshal(body)
+func createSubmitTxBody(arc *ArcClient, tx *broadcast.Transaction, txFormat broadcast.TransactionFormat) ([]byte, error) {
+	body := &SubmitTxRequest{tx.Hex}
+
+	if txFormat == broadcast.RawTxFormat {
+		if err := convertToEfTransaction(arc, body); err != nil {
+			return nil, fmt.Errorf("Conversion to EF format failed: %s", err.Error())
+		}
+	} else if txFormat == broadcast.BeefFormat {
+		// To be implemented
+		return nil, fmt.Errorf("Submitting transactions in BEEF format is unimplemented yet...")
+	}
 
+	data, err := json.Marshal(body)
```

Comment:
I'm not sure about this approach, 
the convertToEfTransaction have side effects, that actually makes it harder to reason about.
Maybe it would be more readable when you would use something like that:
```go
func createSubmitTxBody(arc *ArcClient, tx *broadcast.Transaction, txFormat broadcast.TransactionFormat) (data []byte, err error) {
	var body *SubmitTxRequest
	switch txFormat {
	case broadcast.RawTxFormat:
		body, err = rawTxRequest(tx.Hex)
	case broadcast.EfFormat:
		body, err = efTxRequest(tx.Hex)
	case broadcast.BeefFormat:
		body, err = beefTxRequest(tx.Hex)
	default:
		err = fmt.Errorf("unknown format")
	}
	
	if err != nil {
		return nil, err
	}

	data, err = json.Marshal(body)
	if err != nil {
		return nil, ErrSubmitTxMarshal
	}

	return data, nil
}
```

Reply:
Good point, done.